src/ClimaAtmos/Dycore/src/VanillaAtmosDiscretizations.jl

```diff
@@ -191,41 +195,59 @@ step
 
     This estimate is currently very conservative, needs to be revisited
 """
-function estimatedt(disc::VanillaAtmosDiscretization{T, dim, N, Np, DA},
-                    Q::AtmosStateArray) where {T, dim, N, Np, DA}
+function estimatedt(disc::VanillaAtmosDiscretization{T, dim, N, Np, DA, nmoist},
+                    Q::AtmosStateArray) where {T, dim, N, Np, DA, nmoist}
   @assert T == eltype(Q)
   G = disc.grid
   vgeo = G.vgeo
   # FIXME: GPUify me
   host_array = Array ∈ typeof(Q).parameters
   (h_vgeo, h_Q) = host_array ? (vgeo, Q) : (Array(vgeo), Array(Q))
-  estimatedt(Val(dim), Val(N), G, disc.gravity, h_Q, h_vgeo, G.topology.mpicomm)
+  estimatedt(Val(dim), Val(N), Val(nmoist), G, disc.gravity, h_Q, h_vgeo, G.topology.mpicomm)
 end
 
 # FIXME: This needs cleaning up
-function estimatedt(::Val{dim}, ::Val{N}, G, gravity, Q, vgeo,
-                    mpicomm) where {dim, N}
+function estimatedt(::Val{dim}, ::Val{N}, ::Val{nmoist}, G, gravity, Q, vgeo,
+                    mpicomm) where {dim, N, nmoist}
 
   DFloat = eltype(Q)
 
   Np = (N+1)^dim
   (_, _, nelem) = size(Q)
 
   dt = [floatmax(DFloat)]
+  
+  # Allocate 3 spaces for moist tracers qm, with a zero default value
+  q_m = zeros(DFloat, max(3, nmoist))
 
   if dim == 2
     @inbounds for e = 1:nelem, n = 1:Np
       ρ, U, V = Q[n, _ρ, e], Q[n, _U, e], Q[n, _V, e]
       E = Q[n, _E, e]
       y = vgeo[n, G.yid, e]
-      P = gdm1*(E - (U^2 + V^2)/(2*ρ) - ρ*gravity*y)
-
+      
+      #Compute Temperature and Internal Energy per unit mass
+      #Obtain moist variables from state vector
+      for m = 1:nmoist
+          s = _nstate + m
+          q_m[m] = Q[n, s, e]/ρ
+      end
+      
+        e_int = E / ρ - ((U^2 + V^2)/(2*ρ) + ρ * gravity * y) / ρ
+        T = MoistThermodynamics.saturation_adjustment(e_int, ρ, q_m[1])
+        q_l, q_i = MoistThermodynamics.phase_partitioning_eq(T, ρ, q_m[1])
+        T = MoistThermodynamics.air_temperature(e_int, q_m[1], q_l, q_i)
+      
+      (R_m, cp_m, cv_m, gamma_m) = MoistThermodynamics.moist_gas_constants(q_m[1], q_l, q_i)
+      γ_m =  cp_m/cv_m
       ξx, ξy, ηx, ηy = vgeo[n, G.ξxid, e], vgeo[n, G.ξyid, e],
                        vgeo[n, G.ηxid, e], vgeo[n, G.ηyid, e]
-
-      loc_dt = 2ρ / max(abs(U * ξx + V * ξy) + ρ * sqrt(gamma_d * P / ρ),
-                        abs(U * ηx + V * ηy) + ρ * sqrt(gamma_d * P / ρ))
+      
+       # Calculate local dt
+      loc_dt = 2ρ / max(abs(U * ξx + V * ξy) + ρ * MoistThermodynamics.sound_speed(T, γ_m, R_m),
+                        abs(U * ηx + V * ηy) + ρ * MoistThermodynamics.sound_speed(T, γ_m, R_m)) 
```

Comment:
There is another `soundspeed_air` function in PR #137. That one does not use new fields `gamma_m`, `R_m` but just state variables for which memory is allocated anyway. It might be preferable to re-compute the sound speed, rather than allocating memory for `gamma_m`, `R_m`. @jkozdon? [Just choose which way of calculating sound speed between the two PR’s makes most sense.]